src/TaskLineRenderer.ts

```diff
@@ -31,6 +31,53 @@ export const LayoutClasses: { [c in TaskLayoutComponent]: string } = {
     blockLink: 'task-block-link',
 };
 
+type AttributeValueCalculator = (component: TaskLayoutComponent, task: Task) => string | null;
```

Comment:
Just above here - on lines 22-32 are the following:

```ts
export const LayoutClasses: { [c in TaskLayoutComponent]: string } = {
    description: 'task-description',
    priority: 'task-priority',
    dueDate: 'task-due',
    startDate: 'task-start',
    createdDate: 'task-created',
    scheduledDate: 'task-scheduled',
    doneDate: 'task-done',
    recurrenceRule: 'task-recurring',
    blockLink: 'task-block-link',
};
```

They appear to duplicate data in `FieldLayouts` and slightly feel like they reduce the benefit of this PR....

Also, they appear to suggest the possibility of inconsistent data between these lines and the new code in this PR.

When I comment the lines out, there only seems to be errors in `TaskLineRenderer.test.ts` - which suggests to me that it would be worth trying to adjust the tests to remove the need for this duplicated block of data...

Reply:
Yes indeed, I thought about doing that in a separate PR. I can include it in this one if yo prefer.

Reply:
Actually the change was so easy that I just did that =)